File: server/models/order.js
const {DataTypes, Model} = require('sequelize');
const sequelize = require('../util/database');

class Order extends Model {
}

Order.init({
    // Model attributes are defined here
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true
    },
    itemId: {
        type: DataTypes.INTEGER,
        allowNull: false
        // allowNull defaults to true
    },
    userId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    cartId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    total: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    createdByUserId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    modifiedByUserId: {
        type: DataTypes.INTEGER,
        allowNull: true
    },
    statusId: {
        type: DataTypes.INTEGER,
        allowNull: true
    }
}, {
    createdAt: 'createdDate',
    updatedAt: `modifiedDate`,
    timestamps: true,
    // Other model options go here
    sequelize, // We need to pass the connection instance
    modelName: 'Order' // We need to choose the model name
});

// the defined model is the class itself
console.log(Order === sequelize.models.Order); // true

module.exports = Order;

File: shopping/lib/models/order.dart
import 'dart:io';

class Order {
  int id;
  int itemId;
  int userId;
  int cartId;
  double total;
  int createdByUserId;
  int modifiedByUserId;
  DateTime createdDate;
  DateTime modifiedDate;
  int statusId;

  Order();

  clone() => Order.fromJson(toJson());

  Order.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        itemId = json['itemId'],
        userId = json['userId'],
        cartId = json['cartId'],
        total = json['total'] != null ? json['total'].toDouble() : null,
        createdByUserId = json['createdByUserId'],
        modifiedByUserId = json['modifiedByUserId'],
        createdDate = (json['createdDate'] != null
            ? DateTime.parse(json['createdDate'])
            : null),
        modifiedDate = (json['modifiedDate'] != null
            ? DateTime.parse(json['modifiedDate'])
            : null),
        statusId = json['statusId'];

  Map<String, dynamic> toJson() => {
        'id': id,
        'itemId': itemId,
        'userId': userId,
        'cartId ': cartId,
        'total': total,
        if (createdDate != null) 'createdDate': createdDate.toString(),
        if (modifiedDate != null) 'modifiedDate': modifiedDate.toString(),
        'createdByUserId': createdByUserId,
        'modifiedByUserId': modifiedByUserId,
        if (statusId != null) 'statusId': statusId,
      };
}
